HappyAnding/HappyAnding/Views/SearchView.swift

```diff
@@ -0,0 +1,122 @@
+//
+//  SearchView.swift
+//  HappyAnding
+//
+//  Created by 이지원 on 2022/11/09.
+//
+
+import MessageUI
+import SwiftUI
+
+struct SearchView: View {
+    @Environment(\.isSearching) private var isSearching: Bool
+    @Environment(\.dismissSearch) private var dismissSearch
+    
+    @State var isSearched: Bool = false
+    @State var searchText: String = ""
+    @State var searchResults: [Shortcuts] = []
+    
+    let keywords: [String] = ["단축어", "갓생", "포항꿀주먹"]
+    
+    var body: some View {
+        VStack {
+            if !isSearched {
+                recommendKeword
+                Spacer()
+            } else {
+                if searchResults.count == 0 {
+                    proposeView
+                } else {
+                    ScrollView {
+                        ForEach(searchResults, id: \.self) { result in
+                            NavigationLink(destination: ReadShortcutView(shortcutID: result.id)) {
+                                ShortcutCell(shortcut: result)
+                                    .listRowInsets(EdgeInsets())
+                                    .listRowSeparator(.hidden)
+                            }
+                        }
+                    }
+               }
+            }
+        }
+        .searchable(text: $searchText)
+        .onSubmit(of: .search, runSearch)
+        .onChange(of: searchText) { value in
+            if searchText.isEmpty && !isSearching {
+                // Search cancelled here
+                print("canceled")
+                searchResults.removeAll()
+                isSearched = false
+            }
+        }        .navigationBarTitleDisplayMode(.inline)
+    }
+    
+    private func runSearch() {
+        Task {
+            isSearched = true
+            // firebase 검색 기능 추가해서 searchResults에 넣기
+        }
+    }
+    
+    var recommendKeword: some View {
+        VStack(alignment: .leading) {
+            Text("추천 검색어")
+                .padding(.leading, 16)
+            
+            ScrollView(.horizontal) {
+                HStack {
+                    // TODO: 선택된 텍스트 검색하는 기능
+                    ForEach(keywords, id: \.self) { keyword in
+                        Text(keyword)
+                            .Body2()
+                            .foregroundColor(Color.Gray4)
+                            .padding(10)
+                            .overlay(
+                                RoundedRectangle(cornerRadius: 12)
+                                    .stroke(Color.Gray4, lineWidth: 1)
+                            )
+                    }
+                }
+                .padding(.leading, 16)
+            }
+        }
+    }
+    
+    
+    
+    var proposeView: some View {
+        VStack(alignment: .center) {
+            Text("\(searchText)의 결과가 없어요.\n원하는 단축어를 제안해보는건 어떠세요?").multilineTextAlignment(.center)
+                .Body1()
+                .foregroundColor(Color.Gray4)
+            
+            Button {
+                
```

Comment:
```suggestion
                //TODO: 버튼 클릭 시 단축어 제안하는 페이지 연결
```

추후 기능이 구현되어야하는 부분이라서 주석 추가해두면 좋을 것 같습니다!

Reply:
넹 추가하겠슴다!